hashintel/hash
# H-976: Allow switching pages/entities from public to private

## 🌟 What is the purpose of this PR?




This PR adds the ability to toggle a page or entity to "Public" and "Private".

## 🔗 Related links





- H-976

## 🔍 What does this change?




See commits.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



⚠️ This PR requires backend changes to be applied to main, so using the Vercel deployment risks breaking the production environment

1. Go to a page or an entity, and click "Share" in the top right
2. Toggle the entity to "Public", and refresh the page to ensure it has persisted correctly
3. Toggle back to "Private" and ensure it has persisted correctly

## 📹 Demo



https://github.com/hashintel/hash/assets/42802102/6fc44f49-919e-4b98-aeb0-32c9c657f95c
